docs/design/2022-06-30-partition-table-batchwrite-suppport.md

```diff
@@ -0,0 +1,145 @@
+# Support bypass-TiDB write partition table
+
+- Author(s): [yangxin](http://github.com/xuanyu66)
+- Tracking Issue: https://github.com/pingcap/tispark/issues/2437
+
+## Introduction
+
+This design doc is about how to support bypass-TiDB write partition table with DataSource API.
+
+The semantics to be supported in this feature are:
+
+- INSERT semantics with Data Source API
+- REPLACE semantics with Data Source API
+- DELETE semantics with Spark SQL
+
+And the associated supported partition types to be supported in this feature are:
+
+- [Range partitioning](https://docs.pingcap.com/tidb/dev/partitioned-table#range-partitioning)
+- [Hash partitioning](https://docs.pingcap.com/tidb/dev/partitioned-table#hash-partitioning)
+
+And the parititon expression must be under contions as bellow:
++ column expression
++ `YEAR($argument)` where the argument is a column and its type is datetime or string literal
+that can be parsed as datetime.
+
+## Motivation and background
+
+Partitioning is a way in which a database splits its actual data down into separate tables, but
+still gets treated as a single table by the SQL layer.
+The main goal is to reduce the amount of data read for particular SQL operations so that overall
+response time is reduced while it also can be used to accomplish a number of various objectives.
+
```

Comment:
Plz add a current status description in this section that TiSpark already supports read partition table.